backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,342 @@
+                {
+                    var prevCI = mainCI;
+                    foreach (var follow in followRelation.Follow)
+                    {
+                        // get the predicate remove the first char, first char defines the direction of relation
+                        var predicate = follow.Predicate[1..];
+
+                        // check relation direction
+                        var r = allRelations.Where(r =>
+                        {
+                            if (follow.Predicate[0] == '>')
+                            {
+                                return r.FromCIID == prevCI.ID && r.PredicateID == predicate;
+                            }
+                            else
+                            {
+                                return r.ToCIID == prevCI.ID && r.PredicateID == predicate;
+                            }
+                        }).FirstOrDefault();
+
+
+                        if (r == null)
+                        {
+                            break;
+                        }
+
+                        var targetCI = allCIs.Where(ci =>
+                        {
+                            if (follow.Predicate[0] == '>')
+                            {
+                                return r.ToCIID == ci.ID;
+                            }
+                            else
+                            {
+                                return r.FromCIID == ci.ID;
+                            }
+                        }
+                        ).FirstOrDefault();
+
+                        prevCI = targetCI;
+
+                        var targetCIRequiredTrait = await traitsProvider.GetActiveTrait(follow.RequiredTrait, trans, changesetProxy.TimeThreshold);
+
+                        if (targetCIRequiredTrait == null)
+                        {
+                            logger.LogError($"The required trait {follow.RequiredTrait} for follow realtions was not found!");
+                            continue;
+                        }
+
+                        // check if target CI has the required trait
+                        var aa = await effectiveTraitModel.FilterCIsWithTrait(new List<MergedCI> { targetCI }, targetCIRequiredTrait, layersetVariableRendering, trans, changesetProxy.TimeThreshold);
+
+                        if (aa.ToList().Count == 0)
+                        {
+                            continue;
+                        }
+
+                        var targetCIAttributes = targetCI.MergedAttributes.Where(a => IsAttributeAllowed(a.Value.Attribute.Name, follow.InputWhitelist, follow.InputBlacklist)).ToList();
+
+                        var allRelatetCIAttributes = targetCIAttributes.Select(a => new GatheredAttribute { Name = a.Key, Value = a.Value.Attribute.Value }).ToList();
+
+                        foreach (var mapping in follow.AttributeMapping)
+                        {
+                            foreach (var attribute in allRelatetCIAttributes)
```

Comment:
You have created a list of `GatheredAttributes` from `targetCIAttributes`... why? further below you create new `GatheredAttributes` anyway.